Wikipedia/Code/TalkPageFetcher.swift

```diff
@@ -1,228 +1,55 @@
+    func fetchTalkPageContent(url: URL, completion: @escaping (Result<TalkPageThreadItems?, Error>) -> Void) {
+        guard let pageTitle = url.wmf_title else {
             return
         }
         
-        //todo: should sectionID in CoreData be string?
-        sectionUploader.append(toSection: String(topic.sectionID), text: body, forArticleURL: url) { (result, error) in
-            if let error = error {
-                completion(.failure(error))
-                return
-            }
-            
-            guard let result = result else {
-                completion(.failure(RequestError.unexpectedResponse))
-                return
-            }
-            
-            completion(.success(result))
-        }
-    }
-    
-    func fetchTalkPage(urlTitle: String, displayTitle: String, siteURL: URL, revisionID: Int?, completion: @escaping (Result<NetworkTalkPage, Error>) -> Void) {
-        
-        guard let taskURLWithRevID = getURL(for: urlTitle, siteURL: siteURL, revisionID: revisionID),
-            let taskURLWithoutRevID = getURL(for: urlTitle, siteURL: siteURL, revisionID: nil) else {
-            completion(.failure(RequestError.invalidParameters))
-            return
-        }
-    
-        //todo: track tasks/cancel
-        session.jsonDecodableTask(with: taskURLWithRevID) { (networkBase: NetworkBase?, response: URLResponse?, error: Error?) in
-            
-            if let statusCode = (response as? HTTPURLResponse)?.statusCode,
-                statusCode == 404 {
-                completion(.failure(TalkPageFetcherError.talkPageDoesNotExist))
-                return
-            }
-            
-            if let error = error {
-                completion(.failure(error))
-                return
-            }
-            
-            guard let networkBase = networkBase else {
-                completion(.failure(RequestError.unexpectedResponse))
-                return
-            }
-            
-            //update sort
-            //todo performance: should we go back to NSOrderedSets or move sort up into endpoint?
-            for (topicIndex, topic) in networkBase.topics.enumerated() {
-                
-                topic.sort = topicIndex
+        let params = ["action" : "discussiontoolspageinfo",
+                      "page" : pageTitle,
+                      "format": "json",
+                      "prop" : "threaditemshtml",
+                      "fomatversion" : "2"
+        ]
+
+        performDecodableMediaWikiAPIGET(for: url, with: params) { (result: Result<TalkPageAPIResponse?, Error>) in
+            switch result {
+            case let .success(talk):
+                completion(.success(talk?.threads))
                 
-                for (replyIndex, reply) in topic.replies.enumerated() {
-                    reply.sort = replyIndex
-                }
+            case .failure:
+                completion(.failure(RequestError.invalidParameters))
```

Comment:
In these fetcher classes I like to send up the error in case we want to react to it at a higher level (for example, if it's a network failure, we'll send it back and whoever is calling it can check for that particular error, and fall back to some persisted object for offline mode).

```
            case .failure(let error):
                completion(.failure(error))
```